src/components/Pagination/__tests__/index.test.js

```diff
@@ -0,0 +1,392 @@
+        initialData={Array.from({ length: 50 }, (_, index) => index).map(
+          value => `Item ${value}`,
+        )}
+      >
+        <ExampleChildren />
+      </Pagination>,
+    ))
+
+  test('should render correctly customComponents', async () =>
+    shouldMatchEmotionSnapshot(
+      <Pagination
+        perPage={5}
+        onLoadPage={loadMore}
+        initialData={Array.from({ length: 50 }, (_, index) => index).map(
+          value => `Item ${value}`,
+        )}
+        LoaderComponent={() => <div>Loading</div>}
+        RightComponent={() => "I'm a custom right component"}
+        LeftComponent={() => "I'm a custom left component"}
+        MiddleComponent={({
+          isLoadingPage,
+          page,
+          maxPage,
+          canLoadMore,
+          goToNextPage,
+          goToPage,
+          goToPreviousPage,
+        }) => {
+          const pageNumbersToDisplay = getPageNumbers(page, maxPage, 5)
+          const handlePageClick = pageNumber => () => {
+            goToPage(pageNumber)
+          }
+
+          return (
+            <>
+              <button
+                type="button"
+                disabled={page === 1 || isLoadingPage}
+                onClick={goToPreviousPage}
+              >
+                Back
+              </button>
+              {pageNumbersToDisplay.map(pageNumber => (
+                <button
+                  type="button"
+                  key={pageNumber}
+                  disabled={isLoadingPage}
+                  onClick={handlePageClick(pageNumber)}
+                >
+                  {pageNumber}
+                </button>
+              ))}
+              <button
+                type="button"
+                disabled={(page === maxPage && !canLoadMore) || isLoadingPage}
+                onClick={goToNextPage}
+              >
+                Next
+              </button>
+            </>
+          )
+        }}
+      >
+        <ExampleChildren />
+      </Pagination>,
+    ))
+
+  test('should render correctly with pageClick', async () =>
+    shouldMatchEmotionSnapshot(
+      <Pagination
+        perPage={5}
+        initialData={Array.from({ length: 50 }, (_, index) => index).map(
+          value => `Item ${value}`,
+        )}
+      >
+        <ExampleChildren />
+      </Pagination>,
+      {
+        transform: async node => {
+          const nextButton = node.getByRole('button', { name: 'Next' })
+          const backButton = node.getByRole('button', { name: 'Back' })
+          const firstButton = node.getByRole('button', { name: 'First' })
+          const lastButton = node.getByRole('button', { name: 'Last' })
+          userEvent.click(nextButton)
+          userEvent.click(backButton)
+          userEvent.click(lastButton)
+          userEvent.click(firstButton)
+          const page3Button = node.getByRole('button', { name: 'Page 3' })
+          userEvent.click(page3Button)
+          userEvent.click(page3Button)
```

Comment:
Is it a mistake to make it click twice ?

Reply:
No mistake, to handle when you click on the same page it allow us to don't update the state so no rerender :)